### PR TITLE
Skip clean up `profile.release.package."*"`

### PR DESCRIPTION
**Pull request by hi-rustin**
### What does this PR try to resolve?

close https://github.com/rust-lang/cargo/issues/12620

Skip gc glob profile.

### How should we test and review this PR?

Run the unit test.

### Additional information

None

### CODE REVIEW

File: src/bin/cargo/commands/remove.rs [hunk 1]
```diff
@@ -231,8 +231,13 @@ fn gc_workspace(workspace: &Workspace<'_>) -> CargoResult<()> {
                     package_table.set_implicit(true);
 
                     for (key, item) in package_table.iter_mut() {
+                        let key = key.get();
```
Comment (weihanglo) [inline src/bin/cargo/commands/remove.rs:234]: Can you also fix the comment on line 219, which shouldn't be gc'd?

**Review by weihanglo — APPROVED** [review]
Looks pretty nice. Thank you!

**Comment by hi-rustin** [comment]
Thanks for your review!  💚 💙 💜 💛 ❤️

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks for the fix!

@bors r+

**Comment by weihanglo** [comment]
W've seen this a couple of times (https://github.com/rust-lang/cargo/pull/12575#issuecomment-1697926451, https://github.com/rust-lang/cargo/pull/12317#issuecomment-1611319929, https://github.com/rust-lang/cargo/pull/12539#issuecomment-1688209161). @hi-rustin, are you willing to help fix the flaky test?

Anyway, let @bors retry


<details><summary>See the error log</summary>
<p>

```
---- registry_auth::token_not_logged stdout ----
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: None,
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/3/b/bar",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/dl/bar/1.0.0/download",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
/home/runner/work/cargo/cargo/target/tmp/cit/t2285/dl/bar/1.0.0/download
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/api/v1/crates/new",
        query: None,
        fragment: None,
    },
    method: "put",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/3/f/foo",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
thread 'registry_auth::token_not_logged' panicked at tests/testsuite/registry_auth.rs:499:5:
assertion `left == right` failed
  left: 8
 right: 7
```


</p>
</details>

**Comment by hi-rustin** [comment]
> W've seen this a couple of times ([#12575 (comment)](https://github.com/rust-lang/cargo/pull/12575#issuecomment-1697926451), [#12317 (comment)](https://github.com/rust-lang/cargo/pull/12317#issuecomment-1611319929), [#12539 (comment)](https://github.com/rust-lang/cargo/pull/12539#issuecomment-1688209161)). @hi-rustin, are you willing to help fix the flaky test?

I will take a look later.

**Pull request merged**